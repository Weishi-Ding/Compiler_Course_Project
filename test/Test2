
/*test expression*/
int main(){    
    /*bin op of int*/
    print(2 + 2);
    print(2 - 2);
    print(2 * 2);
    print(2 / 2);
    print(2 == 2);
    print(2 != 2);
    print(2 < 2);
    print(2 <= 2);
    print(2 > 2);
    print(2 >= 2);

    /*bin op of float*/
    print(2.1 + 2.1);
    print(2.1 - 2.2);
    print(2.1 * 2.1);
    print(2.1 / 2.1);
    print(2.1 == 2.1);
    print(2.1 != 2.1);
    print(2.1 < 2.1);
    print(2.1 <= 2.1);
    print(2.1 > 2.1);
    print(2.1 >= 2.1);

    /*unop*/
    print(-2.1);
    print(-10);
}