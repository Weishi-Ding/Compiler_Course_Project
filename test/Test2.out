4
0
4
1
1
0
0
1
0
1
4.200000
-0.100000
4.410000
1.000000
1
0
0
1
0
1
-2.100000
-10
